packages/mobile/src/app/saga.ts

```diff
@@ -207,23 +175,26 @@ function* watchAppState() {
   }
 }
 
-function* handleSetAppState(action: SetAppState) {
+export function* handleSetAppState(action: SetAppState) {
   const appLocked = yield select(getAppLocked)
+  const lastTimeBackgrounded = yield select(getLastTimeBackgrounded)
+  const now = Date.now()
   const cachedPin = getCachedPincode()
   const lockWithPinEnabled = yield select(getLockWithPinEnabled)
-  if (lockWithPinEnabled && action.state === 'background' && !appLocked && !cachedPin) {
-    navigate(Screens.Background, {
-      onUnlock() {
-        navigateBack({ immediate: true })
-      },
-    })
+  if (
+    !cachedPin &&
+    lockWithPinEnabled &&
+    now - lastTimeBackgrounded > DO_NOT_LOCK_PERIOD &&
+    action.state === 'active' &&
+    !appLocked
+  ) {
+    yield put(lock())
```

Comment:
nit, consider giving the `lock` action a more descriptive name

Reply:
Agree, renamed it to `appLock` and `appUnlock`